src/command/backup/file.c

```diff
@@ -294,61 +294,99 @@ backupFile(
                     // Add size filter last to calculate repo size
                     ioFilterGroupAdd(ioReadFilterGroup(readIo), ioSizeNew());
 
-                    // Open the source and destination and copy the file
+                    // Open the source
                     if (ioReadOpen(readIo))
                     {
-                        // Setup the repo file for write. There is no need to write the file atomically (e.g. via a temp file on
-                        // Posix) because checksums are tested on resume after a failed backup. The path does not need to be synced
-                        // for each file because all paths are synced at the end of the backup. It needs to be created in the prior
-                        // context because it will live longer than a single loop when more than one file is being written.
-                        if (write == NULL)
+                        Buffer *const buffer = bufNew(ioBufferSize());
+                        bool readEof = false;
+
+                        // Read the first buffer to determine if the file was truncated. Detecting truncation matters only when
+                        // bundling is enabled as otherwise the file will be stored anyway.
```

Comment:
Why do we still need to save a zero size file when not bundling?
Can't we then restore such a file using the information from the manifest?

Reply:
In some situations, users may do manual restores of their backups, usually because they copied a full backup out of the repo for long term storage. Or, an uncompressed backup can be used directly on ZFS with snapshots. In any case, it would break backward compitility so not something we are going to change. Going forward, most people should use bundling. I wish we could make it the default, but again, backward compatibility.